src/types/dict.rs

```diff
@@ -175,6 +181,12 @@ impl<'a> Iterator for PyDictIterator<'a> {
     }
 }
 
+impl<'py> Drop for PyDictIterator<'py> {
+    fn drop(&mut self) {
+        unsafe { ffi::Py_DECREF(self.dict.as_ptr()) }
```

Comment:
This confuses me: Where have we called INCREF so we can DECREF here?

Reply:
In PyDict::iter.
`to_object` calls `from_borrowed_object`.